src/uu/ls/src/ls.rs

```diff
@@ -733,6 +754,100 @@ fn list(locs: Vec<String>, config: Config) -> i32 {
     }
 }
 
+/// Compare pathbufs in a way that matches the GNU version sort, meaning that
+/// numbers get sorted in a natural way.
+/// One limitation is that the current implementation cannot handle values
+/// greater than u64::MAX.
+fn version_cmp(a: &PathBuf, b: &PathBuf) -> Ordering {
```

Comment:
Would it be possible to add a pure Rust unit tests to cover all cases?
(or you think the current test is good enough?)

Reply:
I believe the tests cover all the cases, or at least the ones I could think of, but it's a little bit tricky to determine what the problem is if it fails. So adding unit tests is a good idea. I'll do that later today.